x-pack/legacy/plugins/apm/public/services/rest/ml.ts

```diff
@@ -32,6 +32,20 @@ interface StartedMLJobApiResponse {
   jobs: MlResponseItem[];
 }
 
+async function getTransactionIndices(http: HttpServiceBase) {
+  const callApmApi: APMClient = createCallApmApi(http);
+  const apmIndicesSettings = await callApmApi({
+    pathname: `/api/apm/settings/ui-indices`
+  });
+  const indexSetting = apmIndicesSettings.find(
+    ({ configuration }) => configuration === 'apm_oss.transactionIndices'
+  );
+  if (!indexSetting) {
+    return 'apm-*';
+  }
```

Comment:
I don't think setting a fallback value is necessary. `apm_oss.transactionIndices` should be guaranteed to be defined.  It's default value is defined in https://github.com/elastic/kibana/blob/ed9aaf1563b0063fdc89d58fc6e76619b95087e4/src/legacy/core_plugins/apm_oss/index.js#L37

Reply:
I took your suggestion and implemented a simple endpoint that returns the configuration. I also simplified the object from a nested object `{apm_oss: {...}}` to a flat object with keys `'apm_oss.transactionIndices', ...`. This simplified the rest APIs a bit. renamed `listUiIndices` to `getApmIndicesSettings` since it's specifically for rendering/editing settings. LMK what you think about these changes.